test/src/bgw/log.c

```diff
@@ -107,6 +116,10 @@ emit_log_hook_callback(ErrorData *edata)
 	}
 	PG_CATCH();
 	{
+		/* If there was an error, rollback what was done before the error */
+		if (IsTransactionState())
+			AbortCurrentTransaction();
```

Comment:
Is that OK do it in the error handler here? Is it possible that `AbortCurrentTransaction()` can also throw an error?

Reply:
It's possible in which case a `RETHROW` might be required to handle it. But that's not needed here.